packages/flutter_tools/lib/src/base/os.dart

```diff
@@ -34,6 +34,28 @@ abstract class OperatingSystemUtils {
   /// if `which` was not able to locate the binary.
   File which(String execName);
 
+  /// Return a list of all paths to `execName` found on the system. Uses the
+  /// PATH environment variable.
+  List<File> whichAll(String execName) {
+    final List<File> result = <File>[];
+    final String pathEnvironmentVariable = platform.environment['PATH'];
+    for (final String pathComponent in _splitPath(pathEnvironmentVariable)) {
+      final String pathToExecName = path.join(pathComponent, execName);
+      if (fs.isFileSync(pathToExecName)) {
```

Comment:
Should this be `processManager.canRun`? canRun takes care of the fact that executables might have various extensions (exe, bat, etc) on Windows.